src/Simplify.elm

```diff
@@ -953,24 +1036,129 @@ errorForUnknownIgnoredConstructor list =
+        | imports =
+            context.imports
+                |> insertImport (import_.moduleName |> Node.value)
+                    { alias =
+                        import_.moduleAlias |> Maybe.map Node.value
+                    , exposed =
+                        case import_.exposingList of
+                            Nothing ->
+                                ExposedSome Set.empty
+
+                            Just (Node _ existingExposing) ->
+                                case existingExposing of
+                                    Exposing.All _ ->
+                                        ExposedAll
+
+                                    Exposing.Explicit list ->
+                                        ExposedSome
+                                            (Set.fromList
+                                                (List.map (\(Node _ expose) -> nameOfExpose expose) list)
+                                            )
+                    }
     }
 
 
+nameOfExpose : Exposing.TopLevelExpose -> String
+nameOfExpose topLevelExpose =
+    case topLevelExpose of
+        Exposing.FunctionExpose name ->
+            name
+
+        Exposing.TypeOrAliasExpose name ->
+            name
+
+        Exposing.InfixExpose name ->
+            name
+
+        Exposing.TypeExpose { name } ->
+            name
+
+
+
+-- DECLARATION VISITOR
+
+
+declarationVisitor : Node Declaration -> ModuleContext -> ModuleContext
+declarationVisitor declarationNode context =
+    let
+        contextReset : ModuleContext
+        contextReset =
+            { context
+                | rangesToIgnore = []
+                , rightSidesOfPlusPlus = []
+                , inferredConstantsDict = RangeDict.empty
+            }
+    in
+    case Node.value declarationNode of
+        Declaration.CustomTypeDeclaration variantType ->
+            let
+                variantNames : Set String
+                variantNames =
+                    variantType.constructors
+                        |> List.map (\(Node _ variant) -> Node.value variant.name)
+                        |> Set.fromList
+
+                variantTypeName : String
+                variantTypeName =
+                    Node.value variantType.name
+            in
+            { contextReset
+                | exposedVariants =
+                    if context.exposedAll then
+                        context.exposedVariants
+                            |> Dict.insert variantTypeName
+                                variantNames
+
+                    else
+                        context.exposedVariants
+                            |> Dict.update variantTypeName
+                                (Maybe.map
+                                    (\_ ->
+                                        variantNames
+                                    )
+                                )
+            }
+
+        _ ->
+            -- no change
+            contextReset
```

Comment:
Actually, we only need this reset only if the declaration is a `FunctionDeclaration`. So we could only change the context in that variant, and use the original context for the type declaration and for the wildcard case.

Reply:
ok